docs/src/main/asciidoc/security-csrf-prevention.adoc

```diff
@@ -9,11 +9,25 @@ include::_attributes.adoc[]
 
 https://owasp.org/www-community/attacks/csrf[Cross-Site Request Forgery (CSRF)] is an attack that forces an end user to execute unwanted actions on a web application in which they are currently authenticated.
 
-Quarkus Security provides a CSRF prevention feature which implements a https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie[Double Submit Cookie technique]. This techninque requires that the CSRF token is never directly exposed to scripts executed on the client-side. In this extension, the CSRF token is:
+Quarkus Security provides a CSRF prevention feature which implements https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie[Double Submit Cookie] and [CSRF Request Header] techniques.
 
-* sent as `HTTPOnly` cookie to the client, and
+`Double Submit Cookie` techninque requires that the CSRF token is never directly exposed to scripts executed on the client-side. CSRF token is:
+
+* sent as `HTTPOnly`, optionally signed, cookie to the client, and
 * directly embedded in a hidden form input of server-side rendered forms, which are transmitted to and used by the client.
 
+`CSRF Request Header` technique can be used to support server-side rendered HTMX or client-side Single-page applications (SPA) which do not use HTML `form` tags.
+
+For the server-rendered pages which do not use an HTML `form` tag, the CSRF token is never directly exposed to scripts executed on the client-side. CSRF token is:
+
+* sent as `HTTPOnly` cookie to the client, and
+* directly added as a custom header value.
+
+For SPAs which do not use an HTML `form` tag, CSRF token is:
+
+* sent as a cookie which can be accessed from JavaScript, users will have to configure the feature to remove an `HttpOnly` property from the cookie
+* read as a cookie value by JavaScript and added as a custom header value.
```

Comment:
I think people will either use templating to set it as a meta tag, or a JS variable, no?

https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#storing-the-csrf-token-value-in-the-dom

```html
<script>
 var csrfToken = "{inject:csrf.token}";
</script>
```

Reply:
@FroMage That is server-rendered, right ? Andy explained that in the SPA case, in order to supply the token, one needs to read the cookie

Reply:
Yes, server-rendered. This works just as well as reading the cookie. If we even need the header to have a token value.

Reply:
@FroMage I've simplified the docs and avoided trying to advice how Java Scripts should handle this token, it is all doc-ed in various farmework's docs